.claude/hooks/pre_tool_use_protect_precommit.py

```
@@ -0,0 +1,189 @@
+    (
+        r"shutil\.(copy|move).*(pyproject\.toml|\.pre-commit-config\.|ruff\.toml|mypy\.ini)",
+        "Script copies/moves config files",
+    ),
+    (
+        r"os\.rename.*(pyproject\.toml|\.pre-commit-config\.|ruff\.toml)",
+        "Script renames config files",
+    ),
+]
+
+
+def is_protected_config_file(file_path) -> bool:
```

Ruff (0.12.2)

Missing type annotation for function argument `file_path`

(ANN001)